### PR TITLE
DashboardPage: Remove file drop code

### PR DESCRIPTION
**Pull request by torkelo**
I get a bit sad when I see one of the most critical components DashboardPage get a bunch of experimental and low-level code added directly to it that has ZERO reasons being inside this component. 

This removes the editPanelCSVDragAndDrop related code which seem to be pretty pointless anyway (at this point) as it's impossible to know you can drop files in this view :) 

Anyway, this needs to be redone properly.

### CODE REVIEW

**Review by oscarkilhed — APPROVED** [review]

**Comment by oscarkilhed** [comment]
I agree, this was an experiment for when the drag and drop was new. This isn't very useful now that the csv upload is more visible in the data source picker.

File: public/app/features/dashboard/containers/DashboardPage.tsx [hunk 1]
```diff
@@ -465,41 +387,13 @@ export class UnthemedDashboardPage extends PureComponent<Props, State> {
...
-              }}
-            </DropZone>
-          ) : (
-            <DashboardGrid
-              dashboard={dashboard}
-              isEditable={!!dashboard.meta.canEdit}
-              viewPanel={viewPanel}
-              editPanel={editPanel}
-            />
-          )}
```
Comment (ivanortegaalba) [inline public/app/features/dashboard/containers/DashboardPage.tsx:502]: I agree this is a complexity we don't want/need to have in a core component. However, removing the feature is too radical for me.

Instead, we could move all the logic of this component into a new component that will enable the feature if the flag is enabled making sure it doesn't generate any performance issue:
```jsx
<DashboardDropZone dashboard={this.props.dashboard}>
  <DashboardGrid
    dashboard={dashboard}
    isEditable={!!dashboard.meta.canEdit}
    viewPanel={viewPanel}
    editPanel={editPanel}
    />
</DashboardDropZone>

// ...
function DashboardDropZone({dashboard, children}){
  if(noFFEnabled){return children;}
  const onFileDrop = ()=>{...}
  return (
    <DropZone
      onDrop={this.onFileDrop}
      accept={DFImport.acceptedFiles}
      maxSize={DFImport.maxFileSize}
      noClick={true}
      >
      ...
        {children}
      ...
    </DropZone>
  )
}

**Review by ivanortegaalba — APPROVED** [review]
Since the feature is not relevant anymore (I double-checked with @oscarkilhed and BI team), I'm happy to remove it. Thanks @torkelo!

File: public/app/features/dashboard/containers/DashboardPage.tsx (hunk 1 above)
Comment (ivanortegaalba) [reply public/app/features/dashboard/containers/DashboardPage.tsx:502]: This is not relevant since the feature can be removed

**Pull request merged**